client/src/components/ChefProfile/ChefSideBar.js

```diff
@@ -95,6 +95,12 @@ const ChefSideBar = (props) => {
   const handleClose = (value) => {
     setOpen(false);
   };
+  useEffect(() => {
+    if (profile && !profile.location) {
+      setControl("EditProfile");
+      setOpen(true);
+    }
+  }, [profile]);
```

Comment:
Might want to add an indicator on the form that location is required so the user doesn't have to try to submit to see what's missing.

Also this maybe not in the scope of this PR, but you could consider adding some additional dialog for onboarding users like "Complete your profile to get started!" so they know why they're being prompted here

Reply:
I added a placeholder *Required for required fields. So that before clicking submit they can see which fields are required.
 But I still need these line of code to prompt the user to set up a profile if they haven't already. The profile here is actually from user model since we made one single model for both user schema and profile and we store profile fields inside the user model. So when the user register they basically have the profile but they don't have the required field. I check here to see if this is the situation, they will be prompt to set up the profile.